src/ItemVirtualMachine.php

```diff
@@ -203,45 +201,48 @@ public static function showForVirtualMachine(CommonDBTM $asset)
             );
 
             if (!empty($hosts)) {
-                echo "<table class='tab_cadre_fixehov'>";
-                echo  "<tr class='noHover'><th colspan='2' >" . __('List of hosts') . "</th></tr>";
-
-                $header = "<tr><th>" . __('Name') . "</th>";
-                $header .= "<th>" . Entity::getTypeName(1) . "</th>";
-                $header .= "</tr>";
-                echo $header;
-
+                TemplateRenderer::getInstance()->display('components/form/item_virtualmachine.html.twig', [
+                    'has_host' => true,
+                ]);
                 $computer = new Computer();
+                $entries = [];
                 foreach ($hosts as $host) {
-                    $class = $host['is_deleted'] ? "deleted" : "";
-                    echo "<tr class='tab_bg_2 $class' >";
-                    echo "<td>";
-                    if ($computer->can($host['computers_id'], READ)) {
-                        echo "<a href='" . Computer::getFormURLWithID($computer->fields['id']) . "'>";
-                        echo $computer->fields['name'] . "</a>";
-                        $tooltip = "<table><tr><td>" . __('Name') . "</td><td>" . $computer->fields['name'] .
-                             '</td></tr>';
-                        $tooltip .= "<tr><td>" . __('Serial number') . "</td><td>" . $computer->fields['serial'] .
-                             '</td></tr>';
-                        $tooltip .= "<tr><td>" . __('Comments') . "</td><td>" . $computer->fields['comment'] .
-                             '</td></tr></table>';
-                        echo "&nbsp; " . Html::showToolTip($tooltip, ['display' => false]);
+                    if ($computer->can($host['items_id'], READ)) {
+                        $entries[] = [
+                            'name' => $computer->getLink($host['items_id']),
+                            'serial' => $computer->fields['serial'],
+                            'comment' => $computer->fields['comment'],
+                            'entity' => Dropdown::getDropdownName('glpi_entities', $computer->fields['entities_id'])
+                        ];
                     } else {
-                        echo $computer->fields['name'];
+                        $entries[] = [
+                            'name' => $computer->fields['name'],
```

Comment:
Should be espaced here due to the usage of the `raw_html` formatter.
```suggestion
                            'name' => htmlspecialchars($computer->fields['name']),
```